### DISCUSSION TITLE
How do I add a feature for recognizing 'frontmatter' metadata in Markdown and Markdown related documents?

### DISCUSSION THREAD

**Discussion by anadon** [jgm/pandoc#9512] [Q&A] [answered]
It seems that Markdown is adding more missing features, in this case free-form metadata.  I've been doing a fair bit of reading the last few days, and it seems that Markdown with 'frontmatter' has been gaining popularity.  

Here is an established example: https://github.com/timlrx/tailwind-nextjs-starter-blog/blob/main/data/blog/code-sample.mdx

Here is a snippet:

```
---
author:
  - name one
  - name two
---
# Start Markdown
```

I would like to add recognition and handling for frontmatter in at least Markdown and ReStructuredText.  How do I do this?

**Comment by anadon** [comment]
https://jekyllrb.com/docs/front-matter/

**Reply by jgm** [reply, marked as answer]
Pandoc has supported this since 2013; indeed, I think we were the first Markdown variant to support this style of metadata.

**Reply by anadon** [reply]
Can you help me with the documentation?  I read over it, but it wasn't apparent to me which extensions need to be enabled.

**Reply by jgm** [reply]
It's enabled by default with `-f markdown`.

**Comment by anadon** [comment]
Thanks for humoring me.  I was having some difficulty with the documentation and API last night.
